Task/2

teuchezh opened the pull request.

teuchezh committed:
fix

teuchezh committed:
fix

teuchezh committed:
Community api

codernoder commented on src/controllers/community-controller.js:
```diff
@@ -1,9 +1,33 @@
 
-    }
+  async createCommunity(req, res, next) {
+    const data = await this.communityService.createCommunity(req);
```
не кидать в сервисы объект запроса. в сервис должны уходить только определённые данные, выбранные заранее

codernoder commented on src/controllers/community-controller.js:
```diff
@@ -1,9 +1,33 @@
+      success: true,
+      data,
+    });
+    next();
```
зачем вызываешь next?

codernoder commented on src/controllers/community-controller.js:
```diff
@@ -1,9 +1,33 @@
+  }
+
+  async communityList(req, res) {
+    const data = await this.communityService.communityList(req);
```
убрать из сервисов объект запроса

codernoder commented on src/services/community-service.js:
```diff
@@ -1,8 +1,27 @@
-const CommunityModel = require('./../db/models/community');
-const CommunityModel = require('../controllers/community-controller');
+const CommunityModel = require('../db/models/community');
+const { collection } = require('../db/models/community');
```
зачем это здесь? нигде не используется

codernoder commented on src/services/community-service.js:
```diff
@@ -1,8 +1,27 @@
   constructor() {}
 
-  //async
+  async createCommunity(req) {
```
убрать объект запроса из сервиса

codernoder commented on src/services/community-service.js:
```diff
@@ -1,8 +1,27 @@
 
-  //async
+  async createCommunity(req) {
+    const com = new CommunityModel({
```
названия переменных аналогичны параметрам request body - используй spread

codernoder commented on src/services/community-service.js:
```diff
@@ -1,8 +1,27 @@
+  }
+
+  async communityList(req) {
+    return await CommunityModel.find({});
```
зачем здесь await?

codernoder commented on src/services/community-service.js:
```diff
@@ -1,8 +1,27 @@
+
+  async community(req) {
+    const id = req.query.id;
+    return await CommunityModel.findById(id).exec();
```
зачем здесь await?

codernoder commented on src/services/community-service.js:
```diff
@@ -1,8 +1,27 @@
+      post_count: req.body.post_count,
+    });
+    com.save();
+    return req.body;
```
зачем возвращать тело запроса? подумай лучше, что здесь нужно возвратить

codernoder commented on src/services/community-service.js:
```diff
@@ -1,8 +1,27 @@
+      sub_count: req.body.sub_count,
+      post_count: req.body.post_count,
+    });
+    com.save();
```
почему не дожидаешься выполнения асинхронной операции?

teuchezh committed:
Subscribe api;
Show subscribed communities - In progress;

teuchezh committed:
API:
Subscribe (/api/users/userapi/subscribe);
Get followers of community (api/collections/community-users);
User generation;
Documentation;
Fixes for requests;
